From 9502326ec4d0f7442fbb3dfc630b30bc3e711a04 Mon Sep 17 00:00:00 2001
From: Jesse Keating <jkeating@redhat.com>
Date: Fri, 24 Jul 2009 16:10:17 -0700
Subject: [PATCH] Handle signing of source rpms.

Source RPM 'arch' in the header can either be noarch, or the arch
of the host preparing the srpm.  Therefor we need to detect a srpm
and react accordingly.
---
 src/server.py |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/src/server.py b/src/server.py
index 4e554bd..9d2bce3 100644
--- a/src/server.py
+++ b/src/server.py
@@ -986,7 +986,13 @@ def cmd_sign_rpm(db, conn):
         field_value = conn.safe_outer_field(field)
         if field_value is None:
             continue
-        rpm_value = hdr[tag]
+        # rpm arch for srpms is noarch or the arch prepared on.  Therefor
+        # check to see if sourcepage is 1
+        if field_value == 'src':
+            if hdr['sourcepackage'] == 1:
+                rpm_value = 'src'
+        else:
+            rpm_value = hdr[tag]
         if rpm_value is None:
             rpm_value = ''
         if field_value != rpm_value:
-- 
1.6.2.5

